x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/action/SetResetModeActionRequest.java

```diff
@@ -22,36 +22,46 @@
 
 public class SetResetModeActionRequest extends AcknowledgedRequest<SetResetModeActionRequest> implements ToXContentObject {
     public static SetResetModeActionRequest enabled() {
-        return new SetResetModeActionRequest(true);
+        return new SetResetModeActionRequest(true, false);
     }
 
-    public static SetResetModeActionRequest disabled() {
-        return new SetResetModeActionRequest(false);
+    public static SetResetModeActionRequest disabled(boolean resetSuccessful) {
+        return new SetResetModeActionRequest(false, resetSuccessful);
     }
 
     private final boolean enabled;
+    private final boolean resetSuccessful;
 
     private static final ParseField ENABLED = new ParseField("enabled");
+    private static final ParseField RESET_SUCCESSFUL = new ParseField("reset_successful");
```

Comment:
I think this should be "delete metadata" or something. Reset successful doesn't make sense in this context, at least not to me.

Especially since you could have `reset_successful` be `true` and have `enabled` to the `true.